yt_dlp/extractor/pietsmiet.py

```diff
@@ -0,0 +1,57 @@
+from .common import InfoExtractor
+from ..utils import (
+    traverse_obj,
+)
+import base64
+
+
+class PietsmietIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?pietsmiet\.de/videos/(?P<id>\d+)-.*'
+    _TESTS = [{
+        'url': 'https://www.pietsmiet.de/videos/77307-brammen-erzahlt-vom-leben-als-geschaftsfuhrer',
+        'md5': '9bca7b377a34bfe4d7a437dd499ff46b',
+        'info_dict': {
+            'id': '77307',
+            'ext': 'mp4',
+            'title': 'Brammen erzählt vom Leben als Geschäftsführer',
+            'thumbnail': 'https://www.pietsmiet.de/storage/media/71/90/03/7161edccc3e3a9f6f78b6d8b637edc2d.jpg'
+        }
+    }]
+
+    # Pietsmiet.de uses base64 to hide a value needed to make API call
+    def _extract_integrity(self, vid):
```

Comment:
This should take the website as a string input instead of saving it as a instance variable